File: lib/dingen_handler_index.erl
-module(dingen_handler_index).

-export([init/3, handle/2, terminate/3]).



init(_Type, Req, _Opts) ->
  {ok, Req, no_state}.


handle(Req, State) ->
  Headers = [{"content-type", "text/html"}],
  Body = "
    <title>Index</title>
    <a href=/hello-world/>hello world</a>
    <br>
    <a href=/hello-random/>hello random</a>
  ",
  {ok, Req2} = cowboy_req:reply(200, Headers, Body, Req),
  {ok, Req2, State}.


terminate(_Reason, _Req, _State) ->
  ok.

File: lib/dingen_handler_index.ex
defmodule Dingen.Handler.Index do
  def init(_type, req, _opts) do
    {:ok, req, :no_state}
  end

  def handle(req, state) do
    headers = [{"content-type", "text/html"}]
    body = "
      <title>Index</title>
      <a href=/hello-world/>hello world</a>
      <br>
      <a href=/hello-random/>hello random</a>
    "
    {:ok, req} = :cowboy_req.reply(200, headers, body, req)
    {:ok, req, state}
  end

  def terminate(_reason, _req, _state) do
    :ok
  end
end
